### PR TITLE
Fix dump of patterns

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Replace node name only if node id is string (it can also be `int` like in patterns)

### Reason for changes

```python
        for original_name in nx_graph.nodes():
>           dot_name = original_name.replace(__CHARACTER_REPLACE_FROM, __CHARACTER_REPLACE_TO)
E           AttributeError: 'int' object has no attribute 'replace'

nncf/common/utils/dot_file_rw.py:104: AttributeError
```

### Tests

tests/common/graph/test_graph_pattern.py::test_dump

### CODE REVIEW

File: tests/common/graph/test_graph_pattern.py [hunk 1]
```diff
@@ -213,3 +213,7 @@ def test_join_pattern_with_special_input_node():
             ref_pattern.add_edge(node, added_node)
 
     assert pattern == ref_pattern
+
+
+def test_dump(tmp_path):
+    TestPattern.first_pattern.dump_graph(tmp_path / "pattern.dot")
```
Comment (kshpv) [inline tests/common/graph/test_graph_pattern.py:219]: What does this test check?

Comment (AlexanderDokuchaev) [reply tests/common/graph/test_graph_pattern.py:219]: That dump_graph function is not failed with AttributeError.
Added check that `pattern.dot` exist.

**Review by kshpv — APPROVED** [review]

File: tests/common/graph/test_graph_pattern.py [hunk 2]
```diff
@@ -213,3 +214,9 @@ def test_join_pattern_with_special_input_node():
             ref_pattern.add_edge(node, added_node)
 
     assert pattern == ref_pattern
+
+
+def test_dump(tmp_path: Path):
+    path_dot = tmp_path / "pattern.dot"
+    TestPattern.first_pattern.dump_graph(path_dot)
+    assert path_dot.exists()
```
Comment (kshpv) [inline tests/common/graph/test_graph_pattern.py:222]: Please remove file after test

**Review by kshpv — CHANGES_REQUESTED** [review]

File: tests/common/graph/test_graph_pattern.py (hunk 2 above)
Comment (AlexanderDokuchaev) [reply tests/common/graph/test_graph_pattern.py]: done

**Review by kshpv — APPROVED** [review]

**Pull request merged**